### PR TITLE
Adds 'fba_code' field to store Amazon FBA Product

### PR DESCRIPTION
**Pull request by riteshshrv**

### CODE REVIEW

**Comment by priyankarani** [comment]
LGTM

File: product.py [hunk 1]
```diff
@@ -160,6 +164,10 @@ class ProductSaleChannelListing:
         'invisible': Eval('channel_source') != 'amazon_mws',
     }, depends=['channel_source'])
 
+    fba_code = fields.Char('Code (Fulfilled By Amazon)', states={
+        'invisible': Eval('channel_source') != 'amazon_mws',
```
Comment (tarunbhardwaj) [inline product.py]: its not in view

File: product.py [hunk 2]
```diff
@@ -46,6 +46,9 @@ class Product:
     gtin = fields.Function(fields.Many2One(
         'product.product.code', 'GTIN'
     ), 'get_codes')
+    fba_code = fields.Function(fields.Many2One(
```
Comment (tarunbhardwaj) [inline product.py]: no need to add here

File: product.py [hunk 3]
```diff
@@ -160,6 +161,11 @@ class ProductSaleChannelListing:
         'invisible': Eval('channel_source') != 'amazon_mws',
     }, depends=['channel_source'])
 
+    fba_code = fields.Char(
+        'Code (Fulfilled By Amazon)',
```
Comment (priyankarani) [inline product.py]: Should be visible only for amazon

File: product.py [hunk 4]
```diff
@@ -147,7 +147,8 @@ def __setup__(cls):
             ('upc', 'UPC'),
             ('isbn', 'ISBN'),
             ('asin', 'ASIN'),
-            ('gtin', 'GTIN')
+            ('gtin', 'GTIN'),
+            ('fba_code', 'Code (Fulfilled By Amazon)')
         ])
```
Comment (priyankarani) [inline product.py]: Remove from here too

**Pull request merged**